core/ide/src/main/java/org/alice/ide/ProjectApplication.java

```diff
@@ -352,20 +363,108 @@ public UserActivity getOpenActivity() {
   }
 
   public final void loadProject(UserActivity activity, UriProjectLoader uriProjectLoader) {
+    loadProject(activity, uriProjectLoader, false, new HashSet<>());
+  }
+
+  public final void loadProject(UserActivity activity, UriProjectLoader uriProjectLoader, boolean isLoadingBackups,
+                                Set<String> unloadableFiles) {
     showWaitCursor();
     this.uriProjectLoader = uriProjectLoader;
     if (uriProjectLoader != null) {
-      uriProjectLoader.deliverContentOnEventDispatchThread(proj -> projectLoaded(activity, proj));
+      uriProjectLoader.deliverContentOnEventDispatchThread(proj -> projectLoaded(activity, proj, isLoadingBackups,
+              unloadableFiles));
     }
   }
 
-  private void projectLoaded(UserActivity activity, Project project) {
+  private void projectLoaded(UserActivity activity, Project project, boolean isLoadingBackups,
+                             Set<String> unloadableFiles) {
+    File saved = UriUtilities.getFile(getUri());
+
+    String parentDirName = "";
+    boolean isBackup = false;
+
+    if (!projectFileUtilities.isNewProject()) {
+      parentDirName = saved.getParentFile().getName();
+      isBackup = BACKUP_EXTENSION.equals(getExtension(parentDirName));
+    }
+
+    File backupDir = null;
+
+    if (isBackup) {
+      backupDir = saved.getParentFile();
+    } else if (saved != null) {
+      backupDir = projectFileUtilities.backupDirectory(saved).toFile();
+    }
+
+    boolean makeVrReady = uriProjectLoader.shouldMakeVrReady();
+
     if (project == null) {
```

Comment:
This is a big method. Consider breaking out the true and false blocks into new methods to make it more readable.